src/game/hud.cpp

```diff
@@ -26,6 +26,8 @@ namespace hud
     FVAR(IDF_PERSIST, hudblend, 0, 1, 1);
     VAR(IDF_PERSIST, hudminimal, 0, 0, 1);
 
+    VAR(IDF_PERSIST, hidechat, 0, 0, 2); // toggle visibility of chat, 1 = hide in-game, 2 hide in-game and in spectator
```

Comment:
Needs a `usage.cfg` entry.

Reply:
Also this should probably be `showchat` instead of `hidechat` to be more in line with variables like `showconsole`.

Reply:
actually I would not do that, since it's purpose is to toggle hiding the chat, not showing the chat

Reply:
Most variables are `show*`. Just make it `1` by default. It's more consistent.

Reply:
I do not know how I would handle that inside cubescript

Reply:
Handle in what way?

Reply:
I would prefer to keep it as "Hide chat" in the menu, but it wouldn't make sense if the variable was called `showchat`.